src/main/scala/ch/epfl/bluebrain/nexus/kg/routes/EventCommonRoutes.scala

```diff
@@ -0,0 +1,79 @@
+package ch.epfl.bluebrain.nexus.kg.routes
+import java.util.UUID
+
+import akka.NotUsed
+import akka.http.javadsl.server.Rejections.validationRejection
+import akka.http.scaladsl.model.headers.`Last-Event-ID`
+import akka.http.scaladsl.model.sse.ServerSentEvent
+import akka.http.scaladsl.server.Directive1
+import akka.http.scaladsl.server.Directives.{optionalHeaderValueByName, provide, reject}
+import akka.persistence.query._
+import akka.persistence.query.scaladsl.EventsByTagQuery
+import akka.stream.scaladsl.Source
+import ch.epfl.bluebrain.nexus.kg.config.AppConfig
+import ch.epfl.bluebrain.nexus.kg.resources.Event
+import io.circe.syntax._
+import io.circe.{Encoder, Printer}
+
+import scala.util.{Failure, Success, Try}
+import scala.concurrent.duration._
+
+/**
+  * Defines commons methods for event routes.
+  */
+trait EventCommonRoutes {
```

Comment:
I would recommend making this an abstract class that takes the required fields in the constructor; this avoids possible initialization issues.